engine/src/main/java/io/zeebe/engine/state/deployment/DeploymentRaw.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.state.deployment;
+
+import io.zeebe.db.DbValue;
+import io.zeebe.msgpack.UnpackedObject;
+import io.zeebe.msgpack.property.ObjectProperty;
+import io.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+
+/** DeplyomentRecord wrapper to store the record in the State. */
+public class DeploymentRaw extends UnpackedObject implements DbValue {
+
+  private final ObjectProperty<DeploymentRecord> deploymentRecordProperty =
+      new ObjectProperty<>("deploymentRecord", new DeploymentRecord());
+
+  public DeploymentRaw() {
+    declareProperty(deploymentRecordProperty);
+  }
+
+  public void setDeploymentRecord(final DeploymentRecord deploymentRecord) {
+    final MutableDirectBuffer valueBuffer = new UnsafeBuffer(0, 0);
+    final int encodedLength = deploymentRecord.getLength();
+    valueBuffer.wrap(new byte[encodedLength]);
+
+    deploymentRecord.write(valueBuffer, 0);
+    deploymentRecordProperty.getValue().wrap(valueBuffer, 0, encodedLength);
+  }
```

Comment:
I see two common patterns for this type of DB wrapper: copy the buffer, or implement a wrap method in the record.

We should choose one pattern and apply this to all DB wrappers. And try to avoid duplication. 
But we can do this in a follow-up issue.

Reply:
https://github.com/zeebe-io/zeebe/issues/6443